compiler/luci/pass/src/RemoveNoEffectSlicePass.cpp

```diff
@@ -0,0 +1,114 @@
+/*
+ * Copyright (c) 2020 Samsung Electronics Co., Ltd. All Rights Reserved
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "luci/Pass/RemoveNoEffectSlicePass.h"
+
+#include <luci/IR/CircleNodes.h>
+
+namespace
+{
+
+/// @brief Return value in CircleConst.
+/// @details Return value in position on CircleConst with int64 format.
+///          Begin must be larger than or equal to 0.
+///          Size must be larger than or equal to -1.
+///          Throw an error on Invalid node or invalid node type.
+int64_t value_from_circle_const(const luci::CircleConst *node, uint32_t idx)
+{
+  if (node->rank() != 1 || node->dim(0).value() <= idx)
+    throw std::runtime_error("Invalid constant node on slice");
```

Comment:
Q) If this is to throw invalid network, I think it should be checked in import stage.

Reply:
Okay. I will remove this shape check and type check below.
I will add a checker on luci/Import

https://github.com/Samsung/ONE/blob/fde0b8aa4d1aa76019b9461370b13e15fc40a9fd/compiler/luci/import/src/Nodes/CircleSlice.cpp#L35